monkey/monkey_island/cc/ui/src/components/configuration-components/ImportConfigModal.tsx

```diff
@@ -36,19 +41,22 @@ const ConfigImportModal = (props: Props) => {
   const authComponent = new AuthComponent({});
 
   useEffect(() => {
-    if (configContents !== null) {
+    if (configContents !== null && configCredentials !== null) {
+      console.log("useEffect was called");
       tryImport();
     }
-  }, [configContents, unsafeOptionsVerified])
+  }, [configContents, configCredentials, unsafeOptionsVerified])
 
   function tryImport() {
```

Comment:
I'm doubtful about the flow control of this function. Have you tested the case where config is encrypted and unsafe options are verified? Does it show the unsafe options modal?

Reply:
Yep, it shows the unsafe options modal.